cornflow-server/cornflow/schemas/user.py

```diff
@@ -66,9 +68,23 @@ class LoginEndpointRequest(Schema):
 class LoginOpenAuthRequest(Schema):
     """
     This is the schema used by the login endpoint with Open ID protocol
+    Validates that either a token is provided, or both username and password are present
     """
 
-    token = fields.Str(required=True)
+    token = fields.Str(required=False)
+    username = fields.Str(required=False)
+    password = fields.Str(required=False)
+
+    @validates_schema
+    def validate_fields(self, data, **kwargs):
+        if data.get("token") is None:
+            if not data.get("username") or not data.get("password"):
+                raise ValidationError(
```

Comment:
This is not 100% true. It could be:
"A token or special user role needs to be provided when using Open ID authentication" or something similar.

Reply:
This error should only pop up if no token is available and either a username or a password is provided, a message and end-user is going to receive as the service user is not going to read them. So for an end-user I prefer to tell them that they need a token.